### PR TITLE
Azure: Create caa image

### PR DESCRIPTION
**Pull request by knrt10**
This adds CAA image for azure provider

Fixes: #226
Signed-off-by: Kautilya Tripathi <ktripathi@microsoft.com>

### CODE REVIEW

**Comment by knrt10** [comment]
cc @bpradipt please review. Also, how do I push to `quay.io/confidential-containers/cloud-api-adaptor-azure`?

**Review by bpradipt — APPROVED** [review]
/lgtm

File: entrypoint.sh [hunk 1]
```diff
@@ -41,6 +41,23 @@ cloud-api-adaptor-aws aws \
...
+  -clientid "${AZURE_CLIENT_ID}" \
+  -secret "${AZURE_SECRET}" \
+  -tenantid "${AZURE_TENANT_ID}" \
+  -region "${AZURE_REGION}" \
+  -instance-size "${AZURE_INSTANCE_SIZE}" \
+  -resourcegroup "${AZURE_RESOURCE_GROUP}" \
+  -subnetid "/subscriptions/${AZURE_SUBSCRIPTION_ID}/resourceGroups/${AZURE_RESOURCE_GROUP}/providers/Microsoft.Network/virtualNetworks/${AZURE_VM_NAME}VNET/subnets/${AZURE_VM_NAME}Subnet" \
+  -securitygroupid "/subscriptions/${AZURE_SUBSCRIPTION_ID}/resourceGroups/${AZURE_RESOURCE_GROUP}/providers/Microsoft.Network/networkSecurityGroups/${AZURE_VM_NAME}NSG" \
+  -imageid "/subscriptions/${AZURE_SUBSCRIPTION_ID}/resourceGroups/${AZURE_RESOURCE_GROUP}/providers/Microsoft.Compute/images/${AZURE_IMAGE}" \
+  -cri-runtime-endpoint /run/containerd/containerd.sock
```
Comment (stevenhorsman) [inline entrypoint.sh:58]: @bpradipt - I've not really seen much of the Azure work, so apologies if this is a silly question, but given the change in #280, is it worth setting `CRI_RUNTIME_ENDPOINT="/run/containerd/containerd.sock"` in `install/overlays/azure/kustomization.yaml`, then swapping `-cri-runtime-endpoint /run/containerd/containerd.sock` for `${optionals}` here to match the other cloud providers?

Comment (bpradipt) [reply entrypoint.sh:58]: Good point @stevenhorsman.
@knrt10 can you please make the changes as suggested by @stevenhorsman to align with other providers ?

Comment (knrt10) [reply entrypoint.sh]: updated. Please review

**Review by stevenhorsman — DISMISSED** [review]
It looks reasonable to me. Thanks for the update!

**Review by stevenhorsman — APPROVED** [review]

**Review by bpradipt — APPROVED** [review]
/lgtm
Thanks @knrt10

**Pull request merged**